### PR TITLE
Add address parent check

### PR DESCRIPTION
**Pull request by valentin-**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | https://github.com/coreshop/CoreShop/issues/1746

Add a check if the parent exists. Tested in /de/shop/customer/addresses and also in the checkout process.

### CODE REVIEW

File: src/CoreShop/Component/Core/Customer/CustomerTransformHelper.php [hunk 1]
```diff
@@ -126,7 +126,7 @@ public function moveAddressToNewAddressStack(AddressInterface $address, ElementI
         }
 
         // set new or changed parent
-        if ($this->isNewEntity($address) === true || $address->getParent()->getId() !== $newParent->getId()) {
+        if ($this->isNewEntity($address) === true || ($address->getParent() && $address->getParent()->getId() !== $newParent->getId())) {
```
Comment (ramundomario) [inline src/CoreShop/Component/Core/Customer/CustomerTransformHelper.php:129]:
```suggestion
        if ($this->isNewEntity($address) === true || $address->getParent()?->getId() !== $newParent->getId()) {
```
We could simplify with a null safe operator, WDYT? :

Comment (valentin-) [reply src/CoreShop/Component/Core/Customer/CustomerTransformHelper.php]: @ramundomario looks good to me, updated.

**Comment by ramundomario** [comment]
PSALM tests failing because of this:
https://github.com/coreshop/CoreShop/blob/19f36e427e424f6bc5523d604c545c0ea9c0361e/src/CoreShop/Component/Resource/Pimcore/Model/PimcoreModelInterface.php#L53-L56
Parent could be null, so PHPDoc should be:
```php
    /**
     * @return ElementInterface|null
     */
```
WDYT @dpfaffenbauer?

**Comment by dpfaffenbauer** [comment]
@ramundomario yes

**Comment by valentin-** [comment]
@ramundomario unfornuately the test still fails https://github.com/coreshop/CoreShop/runs/4656476471?check_suite_focus=true#step:14:42

**Comment by dpfaffenbauer** [comment]
@valentin- That is not your issue now. I will fix the tests and you have to rebase this PR after that.

**Comment by dpfaffenbauer** [comment]
@valentin- can you rebase now pls?

**Comment by valentin-** [comment]
@dpfaffenbauer looks good now.

**Review by dpfaffenbauer — APPROVED** [review]

**Comment by dpfaffenbauer** [comment]
thanks a lot @valentin-

**Pull request merged**